### DISCUSSION TITLE
`compdef: unknown command or service` when installing `OMZP::kubectl`

### DISCUSSION THREAD

**Discussion by hellt** [z-shell/zi#307] [Q&A]
Hi all,
I have been struggling to understand how to handle completions managed by the Oh-My-Zsh plugins.

I am trying to install the `kubectl` plugin like this:

```bash
# some contents in zhrc

# kubectl plugin from ohmyzsh
zi snippet OMZP::kubectl

autoload -Uz compinit
compinit

zi cdreplay -q # <- execute compdefs provided by rest of plugins
zi cdlist
```

I used the `zi zi cdreplay -q` here as suggested in https://wiki.zshell.dev/docs/guides/commands#ignoring-compdefs but when I login to the shell I get

```
-- console output produced during zsh initialization follows --

compdef: unknown command or service: kubectl
compdef: unknown command or service: kubectl
compdef: unknown command or service: kubectl
Recorded compdefs:
compdef kj=kubectl
compdef kjx=kubectl
compdef ky=kubectl
```

`kubectl` is installed at `/usr/local/bin/kubectl` and I get all aliases provided by the OMZP::kubectl but no completions

**Comment by ss-o** [comment]
Hi @hellt :wave: 

Based on the plugin source, a cache variable is expected:

```zsh
typeset ZSH_CACHE_DIR="$ZI[CACHE_DIR]"
```

It should be placed before the `zi snippet OMZP::kubectl` in your `~/.zshrc` file. If you can confirm this solves the issue, I will apply the patch where appropriate.
